static/src/javascripts/projects/common/modules/email/email-article.js

```diff
@@ -0,0 +1,283 @@
+        },
+        headline: 'The headlines, the analysis, the debate',
+        description:
+            'Get the whole picture from a source you trust, emailed to you every morning. The biggest stories examined, and diverse, independent views &ndash; the Guardian Today delivers the best of our journalism.',
+        successHeadline: 'Thank you for signing up to the Guardian Today',
+        successDescription:
+            'We will send you our picks of the most important headlines tomorrow morning.',
+        insertMethod: insertBottomOfArticle,
+    },
+};
+
+const getSpacefinderRules = () => ({
+    bodySelector: '.js-article__body',
+    slotSelector: ' > p',
+    minAbove: 200,
+    minBelow: 150,
+    clearContentMeta: 50,
+    fromBottom: true,
+
+    selectors: {
+        ' .element-rich-link': {
+            minAbove: 100,
+            minBelow: 100,
+        },
+        ' > h2': {
+            minAbove: 200,
+            minBelow: 0,
+        },
+        ' > *:not(p):not(h2):not(blockquote)': {
+            minAbove: 35,
+            minBelow: 200,
+        },
+        ' .ad-slot': {
+            minAbove: 150,
+            minBelow: 200,
+        },
+    },
+});
+
+const addListToPage = (listConfig, successEventName) => {
+    if (listConfig) {
+        listConfig.successEventName = successEventName || '';
+        const iframe = bonzo.create(template(iframeTemplate, listConfig))[0];
+        const $iframeEl = $(iframe);
+        const onEmailAdded = () => {
+            emailRunChecks.setEmailShown(listConfig.listName);
+            session.set('email-sign-up-seen', 'true');
+        };
+
+        bean.on(iframe, 'load', () => {
+            email.init(iframe);
+        });
+
+        if (listConfig.insertMethod) {
+            fastdom.write(() => {
+                listConfig.insertMethod($iframeEl);
+                trackNonClickInteraction(
+                    `rtrt | email form inline | article | ${listConfig.listId} | sign-up shown`
+                );
+                onEmailAdded();
+            });
+        } else {
+            spaceFiller.fillSpace(getSpacefinderRules(), paras => {
+                $iframeEl.insertBefore(paras[0]);
+                trackNonClickInteraction(
+                    `rtrt | email form inline | article | ${listConfig.listId} | sign-up shown`
+                );
+                onEmailAdded();
+            });
+        }
+    }
+};
+
+const init = () => {
+    waitForCheck('emailCanRun')
+        .then(emailCanRun => {
+            if (emailCanRun) {
+                waitForCheck(
+                    'emailCanRunPostCheck'
+                ).then(emailCanRunPostCheck => {
+                    if (emailCanRunPostCheck) {
+                        emailRunChecks
+                            .getUserEmailSubscriptions()
+                            .then(() => {
+                                const [listConfig] =
+                                    Object.entries(
```

Comment:
this is a replacement for https://lodash.com/docs/#find – does it look ok?

Reply:
Concise!

Reply:
Just stepped through this and this all looks ok to me, here's an alternative without the destructuring assignment:

```javascript
const listConfig = Object.keys(listConfigs).find(key => emailRunChecks.listCanRun(listConfigs[key]));
addListToPage(listConfigs[listConfig]);
```